components/compliance-service/reporting/relaxting/util_test.go

```diff
@@ -68,11 +70,52 @@ func TestGetEsIndexWithStartAndEndTime(t *testing.T) {
 func TestFetchLatestDataOrNot(t *testing.T) {
 	filters := make(map[string][]string)
 	filters["start_time"] = []string{"2019-01-02T23:59:59Z"}
-	filters["end_time"] = []string{"2019-01-03T23:59:59Z"}
+	filters["end_time"] = []string{"2019-03-03T23:59:59Z"}
 	latestOnly := FetchLatestDataOrNot(filters)
 	assert.Equal(t, true, latestOnly)
 
 	filters["job_id"] = []string{"job 1"}
 	latestOnly = FetchLatestDataOrNot(filters)
 	assert.Equal(t, false, latestOnly)
 }
+
+func TestGetRepOrSumIndexWithStartTimeAndEndTimeParsingError(t *testing.T) {
+	filters := make(map[string][]string)
+	filters["end_time"] = []string{"2019-03-05T23:59oiZ", "2019-03-05T23:59oiZ"}
+
+	_, err := GetEsIndex(filters, true)
+	assert.Error(t, err, stringError)
+
+	_, err = GetEsIndex(filters, false)
+	assert.Error(t, err, stringError)
+
+}
+
+func TestGetControlIndexWithStartTimeAndEndTime(t *testing.T) {
+	mapping := mappings.IndexNameControl
+
+	filters := make(map[string][]string)
+	filters["end_time"] = []string{"2019-02-03T23:59:59Z", "2019-02-07T23:59:59Z"} // 2019-01-06 should be ignored
+
+	index, err := getControlIndex(filters)
+	assert.Equal(t, mapping+"-2019.02.03*", index)
+	assert.NoError(t, err)
+}
+
+func TestGetControlIndexWithEndTimeWithParsingError(t *testing.T) {
+	filters := make(map[string][]string)
+	filters["end_time"] = []string{"2019-01-05T23:59oiZ", "2019-01-06T23:59oiZ"}
+
+	_, err := getControlIndex(filters)
+
+	assert.Error(t, err, stringError)
+}
+
+func TestGetControlIndexWithStartTimeWithParsingError(t *testing.T) {
+	filters := make(map[string][]string)
+	filters["start_time"] = []string{"2019-01-04T23:59oiZ", "2019-01-06T23:59oiZ"}
+
+	_, err := getControlIndex(filters)
```

Comment:
can we can we test value being returned 
```suggestion
	val, err := getControlIndex(filters)
```
and check assert.Empty(t, val)

Reply:
This is the error test case, we just have to check the error.